framework/wazuh/core/tests/test_requests.py

```diff
@@ -0,0 +1,182 @@
+from datetime import datetime
+from unittest.mock import patch
+from uuid import uuid4
+
+import pytest
+from aiohttp import ClientError
+
+import wazuh
+from wazuh.core.requests import (
+    RELEASE_UPDATES_URL,
+    WAZUH_TAG_KEY,
+    WAZUH_UID_KEY,
+    get_update_information_template,
+    query_update_check_service,
+)
+
+# Fixtures
+
+
+@pytest.fixture
+def client_session_get_mock():
+    with patch('aiohttp.ClientSession.get') as get_mock:
+        yield get_mock
+
+
+@pytest.fixture
+def installation_uid():
+    return str(uuid4())
+
+
+# Tests
+
+@pytest.mark.parametrize('update_check', (True, False))
+@pytest.mark.parametrize('last_check_date', (None, datetime.now()))
+def test_get_update_information_template(last_check_date, update_check):
+    template = get_update_information_template(update_check=update_check, last_check_date=last_check_date)
+
+    assert 'last_check_date' in template
+    assert template['last_check_date'] == (last_check_date if last_check_date is not None else '')
+    assert 'update_check' in template
+    assert template['update_check'] == update_check
+    assert 'current_version' in template
+    assert template['current_version'] == wazuh.__version__
+    assert 'last_available_major' in template
+    assert 'last_available_minor' in template
+    assert 'last_available_patch' in template
+
+
+@pytest.mark.asyncio
+async def test_query_update_check_service_catch_exceptions_and_dont_raise(
+    installation_uid, client_session_get_mock
+):
+    message_error = 'Some client error'
+    client_session_get_mock.side_effect = ClientError(message_error)
+    update_information = await query_update_check_service(installation_uid)
+
+    client_session_get_mock.assert_called()
+
+    assert update_information['status_code'] == 500
+    assert update_information['message'] == message_error
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    'major,minor,patch',
+    (
+        [['5.0.0', '5.0.1'], ['4.9.0', '4.9.1'], ['4.8.1', '4.8.2']],
+        [
+            ['5.0.0', '5.0.1'],
+            ['4.9.0', '4.9.1'],
+            [
+                '4.8.1',
+            ],
+        ],
+        [['5.0.0', '5.0.1'], ['4.9.0'], ['4.8.1', '4.8.2']],
+        [['5.0.0'], ['4.9.1'], ['4.8.1']],
+        [['5.0.0'], ['4.9.1'], []],
+        [['5.0.0'], [], ['4.8.1']],
+        [[], ['4.9.1'], ['4.8.1']],
+        [[], [], []],
+    ),
+)
+async def test_query_update_check_service_returns_correct_data_when_satatus_200(
```

Comment:
```suggestion
async def test_query_update_check_service_returns_correct_data_when_status_200(
```
Also, we should add a docstring for the test.